static/js/publisher/form/fileInput.js

```diff
@@ -11,6 +11,35 @@ class FileInput extends React.Component {
     this.fileClickHandler = this.fileClickHandler.bind(this);
     this.fileChangedHandler = this.fileChangedHandler.bind(this);
     this.keyboardEventHandler = this.keyboardEventHandler.bind(this);
+    this.preventDefaultDragHandler = this.preventDefaultDragHandler.bind(this);
+    this.dragEndHandler = this.dragEndHandler.bind(this);
+    this.dragEnterHandler = this.dragEnterHandler.bind(this);
+    this.dropHandler = this.dropHandler.bind(this);
+
+    this.state = {
+      isDragging: false,
+      canDrop: false
+    };
+  }
+
+  componentDidMount() {
+    document.addEventListener("drag", this.preventDefaultDragHandler);
```

Comment:
Does it need to be in `FileInput` component?
We have it rendered multiple times in a page (5 instances for screenshots). Which means these events are added 5 times to document.

It seems that all we need it one drop target for screenshots, so maybe having these events set by `Media` component would be better? Or does it need access to internals of `FileInput`?

Reply:
The issue is that to enable the "drop here" when you drag onto the window we need to add a listener to the document.

We could have those set by the parent component, but it means `FileInput` isn't self contained. We could remove all previous document event listeners on mount, so we only have 1 set of document eventlisteners (for the global preventDefault stuff) but won't help with the state setting of each component.